stubs/seaborn/seaborn/palettes.pyi

```diff
@@ -0,0 +1,149 @@
+from collections.abc import Iterable, Sequence
+from typing import TypeVar, overload
+from typing_extensions import Literal, Self, TypeAlias
+
+from matplotlib.colors import Colormap, LinearSegmentedColormap, ListedColormap
+from matplotlib.typing import ColorType
+
+__all__ = [
+    "color_palette",
+    "hls_palette",
+    "husl_palette",
+    "mpl_palette",
+    "dark_palette",
+    "light_palette",
+    "diverging_palette",
+    "blend_palette",
+    "xkcd_palette",
+    "crayon_palette",
+    "cubehelix_palette",
+    "set_color_codes",
+]
+
+_T = TypeVar("_T")
+
+SEABORN_PALETTES: dict[str, list[str]]
+MPL_QUAL_PALS: dict[str, int]
+QUAL_PALETTE_SIZES: dict[str, int]
+QUAL_PALETTES: list[str]
+
+class _ColorPalette(list[_T]):
+    def __enter__(self) -> Self: ...
+    def __exit__(self, *args: object) -> None: ...
+    def as_hex(self) -> _ColorPalette[str]: ...
+
+_RGBColorPalette: TypeAlias = _ColorPalette[tuple[float, float, float]]
+_SeabornPaletteName: TypeAlias = Literal[
+    "deep", "deep6", "muted", "muted6", "pastel", "pastel6", "bright", "bright6", "dark", "dark6", "colorblind", "colorblind6"
+]
+
+@overload
+def color_palette(  # type: ignore[misc]
+    palette: _SeabornPaletteName | None = None, n_colors: int | None = None, desat: float | None = None, *, as_cmap: Literal[True]
+) -> list[str]: ...  # this might be a bug in seaborn because we expect the return type to be a Colormap instance
+@overload
+def color_palette(
+    palette: str | Sequence[ColorType], n_colors: int | None = None, desat: float | None = None, *, as_cmap: Literal[True]
```

Comment:
Seems like if `desat` is given (and `as_cmap` is true) it returns a list, not a ColorMap.

Reply:
`desat` and `as_cmap` cannot be used together, you get `TypeError: 'ListedColormap' object is not iterable` at runtime if you pass both.